stock_location_analytic/models/stock_location.py

```diff
@@ -0,0 +1,32 @@
+# Copyright 2017-23 ForgeFlow S.L.
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class StockLocation(models.Model):
+    _inherit = "stock.location"
+
+    analytic_account_id = fields.Many2one(
+        "account.analytic.account", "Analytic Account", copy=False
+    )
+
+    @api.constrains("analytic_account_id")
+    def _check_analytic_account(self):
+        for loc in self:
+            if loc.analytic_account_id:
+                analytic = loc.analytic_account_id
+                if loc.location_id:
+                    parent = loc.location_id
+                    if (
+                        parent.usage != "view"
+                        and parent.analytic_account_id
+                        and parent.analytic_account_id != analytic
+                    ):
+                        raise ValidationError(
+                            _(
+                                """Sublocations can only be
```

Comment:
the message displays ugly:
![image](https://user-images.githubusercontent.com/19620251/223464734-3ec4a696-818c-4d59-94eb-e0c2443fafc1.png)